### PR TITLE
Add extension method RemoveVectorsAtSamePlace

### PR DESCRIPTION
**Pull request by mwojcik-sagiton**
BACKGROUND:
Very often I had to filter vectors list of those which are at the same places. Unfortunately I cannot use `Vector3Comparer` with LINQ `Distinct` method, because it uses `GetHashCode` to determine equality.

DESCRIPTION:
I added new extension method which iterates through list of vectors and checks if any of similar vector is already in list.

TESTING:
Run unit tests

FUTURE WORK:
None

REQUIRED:
- [x] All changes are up to date in CHANGELOG.md.

- - -
This change is [Reviewable](https://reviewable.io/reviews/hypar-io/elements/895)

### CODE REVIEW

**Review by andrewheumann — CHANGES_REQUESTED** [review]

File: Elements/src/Geometry/Vector3Extensions.cs [hunk 1]
```diff
@@ -251,6 +251,27 @@ internal static Vector3 NormalFromPlanarWoundPoints(this IList<Vector3> points)
             }
             return normal.Unitized();
         }
+        
+        /// <summary>
+        /// Remove vectors at the same place
```
Comment (andrewheumann) [inline Elements/src/Geometry/Vector3Extensions.cs:256]: `Remove vectors at the same place` => `De-duplicate a collection of Vectors, such that no two vectors in the result are within tolerance of each other.`

File: Elements/src/Geometry/Vector3Extensions.cs [hunk 2]
```diff
@@ -251,6 +251,27 @@ internal static Vector3 NormalFromPlanarWoundPoints(this IList<Vector3> points)
...
+        /// </summary>
+        /// <param name="vectors">List of vectors</param>
+        /// <param name="tolerance">Distance tolerance</param>
+        /// <returns>List without vectors at the same place</returns>
```
Comment (andrewheumann) [inline Elements/src/Geometry/Vector3Extensions.cs:260]: A new collection of vectors with duplicates removed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@
                                out Dictionary<Guid, List<Polygon>> intersectionPolygons,
                                out Dictionary<Guid, List<Polygon>> beyondPolygons,
                                out Dictionary<Guid, List<Line>> lines)`
+- `IEnumerable<Vector3>.RemoveVectorsAtSamePlace(double tolerance = Vector3.EPSILON)`
```
Comment (andrewheumann) [inline CHANGELOG.md:29]: This belongs in a new section of the changelog at the top, since we released 1.2.0:
```
## 1.3.0

### Added
```

**Review by mwojcik-sagiton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/895)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @andrewheumann)

___
*[`CHANGELOG.md` line 29 at r1](https://reviewable.io/reviews/hypar-io/elements/895#-NDfy1Lf9NpJqKK90AuI-r1-29:-NDlIUEf2hDDlJcjGUQf:b-yzqgqd) ([raw file](https://github.com/hypar-io/elements/blob/29e46deb6cba68f2538187236b9ceb31b2530dba/CHANGELOG.md#L29)):*
<details><summary><i>Previously, andrewheumann wrote…</i></summary><blockquote>

This belongs in a new section of the changelog at the top, since we released 1.2.0:
```
## 1.3.0

### Added
```
</blockquote></details>

indeed you have recently released 1.2.0
___
*[`Elements/src/Geometry/Vector3Extensions.cs` line 256 at r1](https://reviewable.io/reviews/hypar-io/elements/895#-NDfy1Lf9NpJqKK90AuJ-r1-256:-NDlIhbr1zHcpeyqpb65:bb7f2mf) ([raw file](https://github.com/hypar-io/elements/blob/29e46deb6cba68f2538187236b9ceb31b2530dba/Elements/src/Geometry/Vector3Extensions.cs#L256)):*
<details><summary><i>Previously, andrewheumann wrote…</i></summary><blockquote>

`Remove vectors at the same place` => `De-duplicate a collection of Vectors, such that no two vectors in the result are within tolerance of each other.`
</blockquote></details>

done
___
*[`Elements/src/Geometry/Vector3Extensions.cs` line 260 at r1](https://reviewable.io/reviews/hypar-io/elements/895#-NDfy1Lf9NpJqKK90AuJ-r1-260:-NDlIl3V5SPj1brPAjy0:bb7f2mf) ([raw file](https://github.com/hypar-io/elements/blob/29e46deb6cba68f2538187236b9ceb31b2530dba/Elements/src/Geometry/Vector3Extensions.cs#L260)):*
<details><summary><i>Previously, andrewheumann wrote…</i></summary><blockquote>

`List without vectors at the same place` => `A new collection of vectors with duplicates removed.`
</blockquote></details>

done
___
*[`Elements/src/Geometry/Vector3Extensions.cs` line 261 at r1](https://reviewable.io/reviews/hypar-io/elements/895#-NDfy1Lf9NpJqKK90AuJ-r1-261:-NDlIlzf3-htsmW9dHo4:b-n8iz1h) ([raw file](https://github.com/hypar-io/elements/blob/29e46deb6cba68f2538187236b9ceb31b2530dba/Elements/src/Geometry/Vector3Extensions.cs#L261)):*
<details><summary><i>Previously, andrewheumann wrote…</i></summary><blockquote>

Can we rename this to `UniqueWithinTolerance`? To me, "RemoveVectors" suggests it's going to mutate the list it's being called on, and `Vector3`s may not always represent "Places." 

In the future we might consider making `IsAlmostEqualTo` a member of an `IEquatableWithinTolerance` interface, in which case this could become a generic method for any such class. 
</blockquote></details>

I renamed method, thank you for suggestion. 

I like idea of adding new interface, but for now it is fine to remove duplicates only from `Vector3` collection because I found it the most common use case.

**Review by andrewheumann — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/895)* status: :shipit: complete! 1 of 1 approvals obtained

**Pull request merged**